modules/node-shared/src/main/scala/org/tessellation/node/shared/http/routes/SnapshotRoutes.scala

```diff
@@ -40,78 +42,119 @@ final case class SnapshotRoutes[F[_]: Async: Hasher: KryoSerializer, S <: Snapsh
   implicit def binaryAndJsonEncoders[A <: AnyRef: Encoder]: List[EntityEncoder[F, A]] =
     List(BinaryCodec.encoder[F, A], CirceEntityEncoder.circeEntityEncoder[F, A])
 
+  private val serviceUnavailableNodeNotReady: F[Response[F]] =
+    ServiceUnavailable(("message" ->> s"Node should be in Ready state before fetching") :: HNil)
+
+  private val nodeStateF: F[NodeState] = nodeStorage.getNodeState
+  private def validStateForSnapshotReturn(state: NodeState): Boolean = state === NodeState.Ready
+
   protected val httpRoutes: HttpRoutes[F] = HttpRoutes.of[F] {
     case GET -> Root / "latest" / "ordinal" =>
-      import org.http4s.circe.CirceEntityCodec.circeEntityEncoder
-
-      snapshotStorage.headSnapshot.map(_.map(_.ordinal)).flatMap {
-        case Some(ordinal) => Ok(("value" ->> ordinal.value.value) :: HNil)
-        case None          => NotFound()
+      nodeStateF.flatMap { nodeState =>
+        if (validStateForSnapshotReturn(nodeState)) {
```

Comment:
Please avoid naming it `F` at the end. `getNodeState` would be okay IMO. but I think you can also use it directly.
Also, that if statement can be simplified:
```scala
nodeStorage.getNodeState
  .map(validStateForSnapshotReturn)
  .ifM(
    // if true,
    // if false
  )
```

```scala
nodeStorage.getNodeState
  .map(validStateForSnapshotReturn)
  .ifM(
    snapshotStorage.headSnapshot...,
    serviceUnavailableNodeNotReady
  )
```

Reply:
Updated